### PR TITLE
Improve docs for MenuItem Roles

### PR DESCRIPTION
**Pull request by zeke**
I always forget where roles are documented. This PR adds a `### Roles` heading so we can link to the list of roles from elsewhere in the docs. It also cleans up the `Menu` example, tightening the formatting and removing some superfluous `accelerator` and `label` values that didn't need to exist because:

> The `label` and `accelerator` values are optional when using a `role` and will
default to appropriate values for each platform.

cc @bpasero

### CODE REVIEW

**Comment by bpasero** [comment]
Thanks 👍

File: docs/api/menu.md [hunk 1]
```diff
@@ -115,76 +115,36 @@ const template = [
...
-      }
+      {role: 'undo'},
+      {role: 'redo'},
+      {type: 'separator'},
+      {role: 'cut'},
+      {role: 'copy'},
+      {role: 'paste'},
+      {role: 'pasteandmatchstyle'},
+      {role: 'delete'},
+      {role: 'selectall'}
```
Comment (kevinsawicki) [inline docs/api/menu.md]: Much nicer 👍

**Comment by belltown** [comment]
For some reason, when I click on the "roles" link from the electron web site, https://electron.atom.io/docs/api/menu-item/, it takes me to the main Electron documentation page instead of the "roles" section on the MenuItem page.

The link goes to http://electron.atom.io/docs/api/#roles instead of http://electron.atom.io/docs/api/menu-item/#roles.

Note, when viewing the documentation on GitHub, it redirects to the correct link; it just doesn't do it when used from the main web site.

**Comment by kevinsawicki** [comment]
@belltown thanks for reporting this, looks like this issue is affecting several links on the site, would you mind opening an issue on https://github.com/electron/electron.atom.io/issues

**Comment by kevinsawicki** [comment]
@belltown should be fixed by https://github.com/electron/electron.atom.io/pull/725, no need to create a new issue, thanks again for reporting.

**Pull request merged**